vividus/src/main/java/org/vividus/runner/BddStepPrinter.java

```diff
@@ -94,77 +105,105 @@ public static void main(String[] args) throws ParseException, IOException
 
     private static Step createStepFrom(StepCandidate candidate)
     {
-        Step step = new Step();
-        step.setDefinition(candidate.getStartingWord() + SPACE + candidate.getPatternAsString());
+        Step step = new Step(candidate.getStartingWord(), candidate.getPatternAsString());
         Method method = candidate.getMethod();
         // Method could be null for composite steps declared in *.steps files
         boolean compositeInStepsFile = method == null;
         step.setCompositeInStepsFile(compositeInStepsFile);
-        step.setDeprecated(!compositeInStepsFile && method.isAnnotationPresent(Deprecated.class));
+        if (!compositeInStepsFile)
+        {
+            step.setDeprecated(method.isAnnotationPresent(Deprecated.class));
+            String baseFileName = FilenameUtils.getBaseName(
+                    method.getDeclaringClass().getProtectionDomain().getCodeSource().getLocation().toString());
+            step.setLocation(baseFileName.replaceAll("-\\d+\\.\\d+\\.\\d+.*", ""));
+        }
+        else
+        {
+            step.setLocation("");
```

Comment:
empty used two times in the file on lines 118 and 122, shouldn't pmd(or checkstyle I don't remebmer :)) fail the build